app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt

```diff
@@ -126,9 +128,41 @@ class CalendarFragment : FragmentForAccessingTumOnline<EventsResponse>(
     override fun onDownloadSuccessful(response: EventsResponse) {
         isFetched = true
 
-        val events = response.events ?: return
+        var events = response.events ?: return
         scheduleNotifications(events)
 
+        // If the device is in a different timezone, get the daylight savings corrected device offset that is
+        // then used to adjust the event startTime and endTime
+        val deviceTimeZone = TimeZone.getDefault()
+        val germanTimeZone = TimeZone.getTimeZone("Europe/Berlin")
+
+        if(!deviceTimeZone.equals(germanTimeZone)) {
+            // 1000 * 60 * 60, converts from milliseconds to hours, since 10^(-3) seconds * 10^(3) = 1 second
+            //      and there are 60 seconds in a minute and 60 minutes in an hour
+            val millisecondsToHoursFactor = 1000 * 60 * 60
+            val offset = (deviceTimeZone.getOffset(Date().time) - germanTimeZone.getOffset(Date().time)) / millisecondsToHoursFactor
+
+            // A new variable is needed to be able to update event.startTime and event.endTime, since these are
+            // values and changing them to variables will break code in Event.kt
+            var timeZoneCorrectedEvents = emptyList<Event>()
+            for(event in events){
+                timeZoneCorrectedEvents = timeZoneCorrectedEvents.plusElement(Event(
+                        event.description,
+                        // If a null value is present in the database simply propagate it forward
+                        if (event.startTime != null) event.startTime.plusHours(offset) else null,
+                        if (event.endTime != null) event.endTime.plusHours(offset) else null,
+                        event.geo,
+                        event.location,
+                        event.id,
+                        event.status,
+                        event.title,
+                        event.url
+                ))
+            }
+
+            events = timeZoneCorrectedEvents
+        }
+
```

Comment:
The best solution for this might be to change the Event's attributes `startTime` and `endTime` to be an instance of `LocalDateTime` and parse them with German Locale from the API. We'd then be able to just display them in the device's timezone. This might be a little annoying database-wise though.

Another simple solution would be to manipulate the `toCalendarItem` method of the `Event` to convert the times to german local times and then convert them back to dateTimes in the devices timezone like this: 

```Kotlin
val tzGer = DateTimeZone.forID("Europe/Berlin")
val startInDeviceLocation = LocalDateTime(startTime, tzGer).toDateTime(DateTimeZone.getDefault())
val endInDeviceLocation = LocalDateTime(endTime, tzGer).toDateTime(DateTimeZone.getDefault())
```

Reply:
I think I'll go with the second suggestion, unless you would really prefer to go with the first. I missed that there already was a loop going through the events hidden in that method.

I can sadly not test this right now though, because TUMOnline seems to be having issues and wont let me log in with my token.